source/mir/utility/linearfun.d

```diff
@@ -0,0 +1,117 @@
+module mir.utility.linearfun;
+
+import std.traits : isCallable;
+
+/**
+Representation of linear function of the form:
+
+    y = slope * x + intercept
+
+_IMPORTANT_: we can't store the intercept directly as it will lead to
+numerical errors if x and y are similar or equal.
+Hence the representation of a function from the Tinflex paper is used:
+
+    y = slope * (x - y) + a
+
+For a detailed explanation, see https://github.com/libmir/mir/wiki/Numerical-failures
+*/
+struct LinearFun(S)
+{
+    /// direction and steepness
+    S slope; // aka beta
+
+    /// boundary point where f obtains it's maximum
+    S y;
+
+    ///
+    S a;
+
+    ///
+    this(S slope, S y, S a)
+    {
+        this.slope = slope;
+        this.y = y;
+        this.a = a;
+    }
+
+    /// textual representation of the function
+    string toString() const
+    {
+        import std.format: format;
+        import std.math: abs, isNaN;
+        char sgn = intercept > 0 ? '+' : '-';
+        if (slope.isNaN)
+            return "#NaN#";
+        else
+            return format("%.2fx %c %.2f", slope, sgn, abs(intercept));
+    }
```

Comment:
@9il this function is nice for debugging, however it has the downside that if one wants to copy the generated intervals this string is printed by default instead of e.g. `LinearFun!double(0.159263, -1.36003, 1.26786)`, so I am inclined to remove or rename this function. What's your opinion?